test/ecs.go

```diff
@@ -0,0 +1,117 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build integration
+// +build integration
+
+package test
+
+import (
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"strings"
+)
+
+func RestartDaemonService(clusterArn *string, serviceName *string) error {
+	return RestartService(clusterArn, nil, serviceName)
+}
+
+func RestartService(clusterArn *string, desiredCount *int64, serviceName *string) error {
+	sess := session.Must(session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	}))
+
+	svc := ecs.New(sess)
+
+	forceNewDeployment := true
+
+	updateServiceInput := &ecs.UpdateServiceInput{
+		Cluster:            clusterArn,
+		Service:            serviceName,
+		ForceNewDeployment: &forceNewDeployment,
+	}
+	if desiredCount != nil {
+		updateServiceInput = updateServiceInput.SetDesiredCount(*desiredCount)
+	}
+
+	_, err := svc.UpdateService(updateServiceInput)
+
+	return err
+}
+
+type ContainerInstance struct {
+	ContainerInstanceArn string
+	ContainerInstanceId  string
+	EC2InstanceId        string
+}
+
+func GetContainerInstances(clusterArn *string) ([]ContainerInstance, error) {
+	containerInstanceArns, err := GetContainerInstanceArns(clusterArn)
+	if err != nil {
+		return []ContainerInstance{}, err
+	}
+
+	describeContainerInstancesOutput, err := describeContainerInstances(clusterArn, containerInstanceArns)
+	if err != nil {
+		return []ContainerInstance{}, err
+	}
+
+	results := []ContainerInstance{}
+	for _, containerInstance := range describeContainerInstancesOutput.ContainerInstances {
```

Comment:
Would it better to have an internal that calls [IMDS (from EC2)](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html) and also [metadata from ECS](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint-v4.html) so it would better to have less specific calls ?

Reply:
These tests don't run on container instances. Agent runs there but tests execute from github's own runners.

Reply:
We can run that. However, its would not be the same case for Fargate (but based on the aim for this PR, should we only focus on EC2).

Reply:
1. We can't know exact ec2 instances running without calling this describecontainerinstances, if using autoscaling (which this test is)
2. metadata from ecs thingy seems to be callable using environment variable within each container/task, which aren't accessible in github's host?